internal/server/providers/oidc_test.go

```diff
@@ -2,16 +2,850 @@ package providers
+		"errorId": "aaabbb",
+		"errorCauses": []
+	}`
+
+	//nolint:gosec // this is not an actual secret credential
+	oktaInvalidClientSecretResp = `{
+		"error": "invalid_client",
+		"error_description": "The client secret supplied for a confidential client is invalid."
+	}`
+
+	oktaInvalidAuthCodeResp = `{
+		"error": "invalid_grant",
+		"error_description": "The authorization code is invalid or has expired."
+	}`
+)
+
+func setupDB(t *testing.T) *gorm.DB {
+	driver, err := data.NewSQLiteDriver("file::memory:")
+	assert.NilError(t, err)
+
+	patch.ModelsSymmetricKey(t)
+	db, err := data.NewDB(driver, nil)
+	assert.NilError(t, err)
+
+	return db
+}
+
+func (ts *testOIDCServer) run(t *testing.T) string {
+	newMux := http.NewServeMux()
+	server := httptest.NewTLSServer(newMux)
+
+	// this is the jwks public key that corresponds with _testdata/test-server-sec.key we just loaded
+	jwks := `{
+		"keys": [
+			{
+				"kty": "RSA",
+				"n": "17B6CrEmJYA2-bPUxaL0OqBg3PWs19ab2DIwrgNc2vPdseV2rUWtXTjVFhVM4cfhuJRULpGJHw8Us3YyJNrIWOaPhOlLCOkrtWK3jCl2xEhgLkpyGxLzsjkSGmOmTNtz44dnxW3WNW-3fKr-aAlGI9O81x3tbU7NnNvvFkotC2HvElQIjISw118C13b9SY1Xc7iKJYcwu6NiSnnkazGGkDLZcR0Ja4lp7Iym8sIbQ72o_mGUUDoCO3CDJePliq4RRKvxCPH2SOuzRpUGwzyK4bfZPXRR17GcOd7DJU6kfGxfX6oAuJCqCEUZCYfQYo6Uj4cBl0BPANlPQuJk7irRvw",
+				"e": "AQAB",
+				"alg": "RS256",
+				"kid": "openid-test",
+				"use": "sig"
+			}
+		]
+	}`
+
+	wellKnown := fmt.Sprintf(`{
+		"issuer": "%[1]s",
+		"authorization_endpoint": "%[1]s/auth",
+		"token_endpoint": "%[1]s/token",
+		"jwks_uri": "%[1]s/keys",
+		"userinfo_endpoint": "%[1]s/userinfo",
+		"id_token_signing_alg_values_supported": ["RS256"]
+	}`, server.URL)
+
+	// general OIDC endpoints
+	newMux.HandleFunc("/.well-known/openid-configuration", func(w http.ResponseWriter, req *http.Request) {
+		_, err := io.WriteString(w, wellKnown)
+		if err != nil {
+			w.WriteHeader(500)
+		}
+	})
+	newMux.HandleFunc("/keys", func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Add("Content-Type", "application/json")
+		_, err := io.WriteString(w, jwks)
+		if err != nil {
+			w.WriteHeader(500)
+		}
+	})
+	newMux.HandleFunc("/token", func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Add("Content-Type", "application/json")
+		w.WriteHeader(ts.tokenResponse.code)
+		_, err := io.WriteString(w, ts.tokenResponse.body)
+		if err != nil {
+			w.WriteHeader(500)
+		}
+	})
+	newMux.HandleFunc("/userinfo", func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Add("Content-Type", "application/json")
+		_, err := io.WriteString(w, ts.userInfoResponse)
+		if err != nil {
+			w.WriteHeader(500)
+		}
+	})
+
+	// azure endpoints
+	newMux.HandleFunc("/v1.0/me/memberOf/fail", func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(403)
+	})
+	newMux.HandleFunc("/v1.0/me/memberOf", func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Add("Content-Type", "application/json")
+		_, err := io.WriteString(w, azureGroupResponse)
```

Comment:
We should probably only add provider specific endpoints when testing that one provider. Otherwise we risk the tests of other providers not being exactly right.

Adding these for only the azure tests should be easy by accepting a func to add additional handlers:

```go
func (t *testing.T, mux *http.ServeMux) {
    ...
}
```

I made those changes in a1dde1bfa8222010e9a6a63052afe7de9cdbf388 if you want to check it out, or cherry-pick it to apply the changes.